Add Media Player component for Vizio Sound Bars

raman325 opened the pull request:
## Description: I created a new media player component to support Vizio Sound Bars. It's functionally very similar to the Vizio TV component, the only difference is there is no pairing or auth process.

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#8915

## Example entry for `configuration.yaml` (if applicable):
```yaml
media_player:
  - platform: viziosoundbar
    suppress_warning: 'True'
    name: Sound Bar
    host: <HOST_IP>
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23

raman325 committed:
Added Media Player component for Vizio Sound Bars

raman325 committed:
Update requirements to include pyviziosoundbar

raman325 committed:
Reference viziosoundbar component in .coveragerc

homeassistant commented:
Hi @raman325, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#21950).

Once you do that we will be able to review and accept this pull request.

Thanks!

raman325 committed:
Update power on and off method calls

raman325 commented:
I'm not sure why the build failed this second time, it worked the first time except I referenced invalid method names for powering on and off the device. I fixed that, and the logs for the second run are referencing things I did not touch. Can someone advise?

I'm also not sure why cla-bot and Hound are still in pending/expected status, they cleared pretty quickly the first time and nothing I changed should impact either check.

Finally, this would be a breaking change but it may be worth considering changing the vizio component to viziotv if this component gets approved and merged. The base package for the vizio component (pyvizio) assumes that the device has to go through the pairing process in order to get an auth token and all of the methods require auth. Sound Bars do not need either of these, but it seemed easier to create a separate python package and component rather than try to overload the single package/component for both use cases

balloob reviewed (changes requested):
You need to get your changes in pyviziosoundbar merged into pyvizio. You are now just forking it, that is not allowed. When you have merged your changes into pyvizio, update the vizio platform to add support for soundbars.

References:
 - https://github.com/vkorn/pyvizio/tree/master/pyvizio
 - https://github.com/raman325/pyviziosoundbar/tree/master/pyviziosoundbar
